apps/extension/src/core/domains/accounts/handler.ts

```diff
@@ -124,30 +103,41 @@ export default class AccountsHandler extends ExtensionHandler {
     const mnemonic = splitIdx === -1 ? suri : suri.slice(0, splitIdx)
     const derivationPath = splitIdx === -1 ? "" : suri.slice(splitIdx)
 
-    let derivedMnemonicId = await this.stores.seedPhrase.getExistingId(mnemonic)
+    const meta: KeyringPair$Meta = {
+      name,
+    }
 
-    if (!derivedMnemonicId) {
-      const result = await this.stores.seedPhrase.add(
-        `${name} Recovery Phrase`,
-        mnemonic,
-        password,
-        SOURCES.Imported,
-        true
-      )
-      if (result.ok) derivedMnemonicId = result.val
-      else throw new Error("Failed to store mnemonic", { cause: result.val })
+    // suri could be a private key instead of a mnemonic
+    if (mnemonicValidate(mnemonic)) {
+      const derivedMnemonicId = await this.stores.seedPhrase.getExistingId(mnemonic)
+
+      if (derivedMnemonicId) {
+        meta.origin = AccountTypes.SEED
+        meta.derivedMnemonicId = derivedMnemonicId
+        meta.derivationPath = derivationPath
+      } else {
+        const result = await this.stores.seedPhrase.add(
+          `${name} Recovery Phrase`,
+          mnemonic,
+          password,
+          SOURCES.Imported,
+          true
+        )
+        if (result.ok) {
+          meta.origin = AccountTypes.SEED // find a better name
+          meta.derivedMnemonicId = result.val
+          meta.derivationPath = derivationPath
+        } else throw new Error("Failed to store mnemonic", { cause: result.val })
+      }
+    } else {
+      meta.origin = AccountTypes.SEED // TODO "LOCAL"
```

Comment:
In the case that `mnemonicValidate` is false, we are assuming that suri is a valid private key. Should we also validate that to ensure it isn't just garbage?

Reply:
I think no need, if it was invalid the keyring.addUri call could break

Reply:
![image](https://github.com/TalismanSociety/talisman/assets/26880866/826f4389-c448-4ecc-a4eb-0d43e775bd92)